src/action/base/create_or_merge_nix_config.rs

```diff
@@ -0,0 +1,490 @@
+            self.tracing_synopsis(),
+            vec![format!(
+                "If {} already exists, we will attempt to merge the current settings with our settings; \
+                otherwise, it will be created with only our settings",
+                self.path.display()
+            )],
+        )]
+    }
+
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn execute(&mut self) -> Result<(), ActionError> {
+        let Self { path, nix_configs } = self;
+
+        if tracing::enabled!(tracing::Level::TRACE) {
+            let span = tracing::Span::current();
+            span.record(
+                "merged_nix_config",
+                nix_configs
+                    .merged_nix_config
+                    .settings()
+                    .iter()
+                    .map(|(k, v)| format!("{k}='{v}'"))
+                    .collect::<Vec<_>>()
+                    .join(" "),
+            );
+        }
+
+        let mut options = OpenOptions::new();
+        options.create(true).write(true).read(true);
+        options.mode(NIX_CONF_MODE);
+
+        // Create a temporary file in the same directory as the one
+        // that the final file goes in, so that we can rename it
+        // atomically
+        let parent_dir = path.parent().expect("File must be in a directory");
+        let mut temp_file_path = parent_dir.to_owned();
+        {
+            let mut rng = rand::thread_rng();
+            temp_file_path.push(format!("nix-installer-tmp.{}", rng.gen::<u32>()));
+        }
+        let mut temp_file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            // If the file is created, ensure that it has harmless
+            // permissions regardless of whether the mode will be
+            // changed later (if we ever create setuid executables,
+            // they should only become setuid once they are owned by
+            // the appropriate user)
+            .mode(0o600)
+            .open(&temp_file_path)
+            .await
+            .map_err(|e| {
+                ActionError::Open(temp_file_path.clone(), e)
+            })?;
+
+        // FIXME(@cole-h): for now we replace the entire file, but in the future we could potentially "replace" the contents
+        let mut new_config = String::new();
+        if let Some(existing_nix_config) = &nix_configs.existing_nix_config {
+            for (name, value) in existing_nix_config.settings() {
+                if nix_configs.merged_nix_config.settings().get(name).is_some() {
+                    continue;
+                }
+
+                new_config.push_str(name);
+                new_config.push_str(" = ");
+                new_config.push_str(value);
+                new_config.push('\n');
+            }
+
+            new_config.push('\n');
+        }
+
+        new_config.push_str(&format!(
+            "# Generated by https://github.com/DeterminateSystems/nix-installer, version {version}.\n",
```

Comment:
This may not actually be correct anymore. I wonder if we should have this at all now...

Reply:
Well, it would be correct if we weren't curing, but yeah... I'm open to either-or.

FWIW, if the user already has all of the settings we want, this won't appear at all. Only if we had to change any settings will it show up.

Reply:
Let's leave it as-is and if we get feedback we can change it.